src/base.ts

```diff
@@ -29,6 +33,91 @@ export abstract class Base {
     this.baseUrl = "http://localhost:3000";
     this.disabledDataSettings = DisableDataSettings.None;
     this.enabled = true;
+    this.appDetails = {
+      platform_id: null,
+      app_version: null,
+      server_version: null,
+      store_id: null,
+      source_id: null,
+    };
+    this.anonId = this.generateAnonId()
+    this.userDetails = {
+      user_id: this.anonId,
+      email: null,
+      wallet: null
+    }
+  }
+
+  protected generateAnonId(): any {
+    //todo:paul: get anon hash from @christian
+    let hash = 'anon_example123'
+    if (ExecutionEnvironment.canUseDOM) {
+      let storedHash = localStorage.getItem('helikaAnonId');
+      if (storedHash && !_.isEmpty(storedHash)) {
+        hash = storedHash;
+      } else {
+        localStorage.setItem('helikaAnonId', hash)
+      }
+    }
+    return hash
+  }
+
+  protected getUserDetails(): any {
+    return this.userDetails;
+  }
+
+  protected setUserDetails(details: {
+    user_id: string,
+    email?: string | undefined,
+    wallet?: string | undefined,
+    [key: string]: any;
+  }): any {
+    this.userDetails = Object.assign({}, this.userDetails, details)
+  }
+
+  protected getAppDetails(): any {
+    return this.appDetails;
+  }
+
+  protected setAppDetails(details: {
+    platform_id?: string | undefined,
+    app_version?: string | undefined,
+    store_id?: string | undefined,
+    source_id?: string | undefined,
+    server_version?: string | undefined,
+    [key: string]: any;
+  }): any {
+    this.appDetails = Object.assign({}, this.appDetails, details)
+  }
+
+  protected getDeviceDetails(): any {
```

Comment:
DeviceDetails() probably doesn't change at any time. you can probably call setDeviceDetails() on init, save as private static, and then return it here getDeviceDetails();